spec/services/record_declarations/started/early_career_teacher_spec.rb

```diff
@@ -2,51 +2,290 @@
+    end
+
+    it "raises an error" do
+      expect {
+        subject.call
+      }.to raise_error(ActionController::ParameterMissing, /The property '#\/declaration_date' can not declare a future date/)
+    end
+  end
+
+  context "when declaration_date before milestone start" do
+    let(:declaration_date) { (Finance::Milestone.pluck(:start_date).min - 10.days).rfc3339 }
 
-  let(:cutoff_start_datetime) { ect_profile.schedule.milestones.find_by(declaration_type: "started").start_date.beginning_of_day }
-  let(:cutoff_end_datetime) { ect_profile.schedule.milestones.find_by(declaration_type: "started").milestone_date.end_of_day }
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+    end
 
-  before do
-    travel_to cutoff_start_datetime + 2.days
-    create(:ecf_statement, :output_fee, deadline_date: 6.weeks.from_now, cpd_lead_provider:)
+    it "raises an error" do
+      expect {
+        subject.call
+      }.to raise_error(ActionController::ParameterMissing, /The property '#\/declaration_date' can not be before milestone start/)
+    end
   end
 
-  it_behaves_like "a participant declaration without evidence held service" do
-    def given_params
-      ect_params
+  context "when declaration_date after milestone end" do
+    let(:declaration_date) { Date.new(2021, 12, 1).rfc3339 }
+
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
     end
 
-    def given_profile
-      ect_profile
+    it "raises an error" do
+      expect {
+        travel_to Date.new(2021, 12, 20)
+
+        subject.call
+
+        travel_back
+      }.to raise_error(ActionController::ParameterMissing, /The property '#\/declaration_date' can not be after milestone end/)
     end
   end
 
-  it_behaves_like "a participant service for ect" do
-    def given_params
-      ect_params
+  context "when user profile ParticipantProfileState is withdrawn, but was active on declaration date" do
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+      create(:participant_profile_state, :withdrawn, participant_profile: profile, cpd_lead_provider:)
     end
 
-    def user_profile
-      ect_profile
+    it "creates the declaration" do
+      expect { subject.call }.to change { ParticipantDeclaration.count }.by(1)
     end
   end
 
-  context "when user is for 2020 cohort" do
-    let!(:cohort_2020) { create(:cohort, start_year: 2020) }
-    let!(:school_cohort_2020) { create(:school_cohort, cohort: cohort_2020, school: ect_profile.school) }
+  context "when user profile ParticipantProfileState is withdrawn and also withdrawn on declaration date" do
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+      create(:participant_profile_state, :withdrawn, participant_profile: profile, cpd_lead_provider:, created_at: declaration_date_object - 2.days)
```

Comment:
can you use the service class + travel_to to do this ?

Reply:
which service class are you referring to?

Reply:
`Participants::Withdraw::(ECT|Mentor)`

Reply:
done